spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-stream-functional-binder-sample/README.adoc

```diff
@@ -0,0 +1,30 @@
+
+The sample consists of two applications, a source (`spring-cloud-gcp-pubsub-stream-functional-binder-sample-source`) and a sink (`spring-cloud-gcp-pubsub-stream-functional-binder-sample-sink`). The source collects a user name and a message through a web form, adds the data along with the time of message creation to a `UserMessage` object, and sends it through Google Cloud Pub/Sub. The sink application simply logs the message.
+
+If the topic for the sink and source does not exist, the binder will automatically create them in Google Cloud Pub/Sub based on the values in link:spring-cloud-gcp-pubsub-stream-functional-binder-sample-source/src/main/resources/application.properties[source] and link:spring-cloud-gcp-pubsub-stream-functional-binder-sample-sink/src/main/resources/application.properties[sink] application.properties.
+
+If you would like to implement load balancing with multiple subscribers reading from the same subscription, you can specify a custom group in the sink's link:spring-cloud-gcp-pubsub-stream-functional-binder-sample-sink/src/main/resources/application.properties[application.properties]. If the group is not specified, an anonymous subscription with the name `anonymous.<topicName>.<randomUUID>` will be generated.
+
+== Setup & Configuration
+
+1. Configure your credentials and project ID by following link:../../docs/src/main/asciidoc/core.adoc#project-id[these instructions].
++
+Alternatively, if you have the https://cloud.google.com/sdk/[Google Cloud SDK] installed and initialized, and are logged in with https://developers.google.com/identity/protocols/application-default-credentials[application default credentials], Spring will auto-discover those parameters for you.
+
+2. Set your project ID using the `spring.cloud.gcp.project-id` property in link:src/main/resources/application.properties[application.properties] or use the `gcloud config set project [PROJECT_ID]` Cloud SDK command.
+
+== Running the code sample
+
+In the link:src/main/resources/application.properties[application.properties] file, a topic and group name are already preconfigured for you (it's `my-topic` and `my-group`, so the subscription name used by the `Sink` is `my-topic.my-group`).
+The topic and subscription will be created in your account if they do not already exist.
+
+1. In separate terminal windows, start the `spring-cloud-gcp-pubsub-stream-functional-binder-sample-source` and `spring-cloud-gcp-pubsub-stream-functional-binder-sample-sink` applications with the `$ mvn spring-boot:run` command in the same folder as the apps' `pom.xml` files.
+
+5. Browse to the source application at `http://localhost:8080`, type in a message and a user name, and press the `Post it!` button.
+
+6. Verify in the `spring-cloud-gcp-pubsub-stream-functional-binder-sample-sink` app's logs that the message was received:
++
+`New message received from [USERNAME] at [TIME_MESSAGE_SENT]: [MESSAGE]`
```

Comment:
Do we need a cleanup step?
Will those auto-created topics and subscriptions just hang around?

Reply:
Auto-created anonymous subscriptions get cleaned up, but manually specified topics and topic.group subscriptions stick around. I'll add a cleanup step.